tensorflow/core/kernels/stack.cc

```diff
@@ -319,6 +419,40 @@ void StackPopOp::ComputeAsync(OpKernelContext* ctx, DoneCallback done) {
     ctx->set_output(0, value.tensor);
     done();
   }
+
+  // This is the reverse of the heuristic for swapping out, asynchronously
+  // swap in most recent swapped tensors if the GPU allocator says less than
+  // kOccupancy of the memory is in use.
+  static constexpr double kOccupancy = 0.7;
+
+  DeviceContext* device_ctxt = ctx->op_device_context();
+  Device* device = static_cast<Device*>(ctx->device());
+  Allocator* gpu_allocator = device->GetAllocator(value.alloc_attrs);
+  absl::optional<AllocatorStats> stats = gpu_allocator->GetStats();
+  if (!stats || !*stats->bytes_limit ||
+      stats->bytes_in_use > (*stats->bytes_limit * kOccupancy)) {
+    return;
+  }
+
+  // Obtain the most recent swapped TensorAndAllocation.
+  Stack::TensorAndAllocation* to_swap_in = nullptr;
+  size_t to_swap_in_index;
+  if (!stack->GetTensorToSwapIn(&to_swap_in, &to_swap_in_index)) {
+    return;
+  }
+
+  Tensor* device_tensor =
+      new Tensor(gpu_allocator, (to_swap_in->tensor).dtype(),
+                 (to_swap_in->tensor).shape());
+  device_ctxt->CopyCPUTensorToDevice(
+      &(to_swap_in->tensor), device, device_tensor,
+      [stack, to_swap_in, device_tensor, to_swap_in_index](const Status& s) {
+        if (s.ok()) {
+          stack->MarkAsUnswapped(to_swap_in, device_tensor);
```

Comment:
` error: ignoring return value of function declared with 'warn_unused_result' attribute [-Werror,-Wunused-result]`

Reply:
Same as the above.